doc/developer-guides/coding_guidelines.rst

```diff
@@ -1787,6 +2323,197 @@ Compliant example::
        }
 
 
+TY-21: The bit-filed shall either be signed integer or unsigned integer or bool
```

Comment:
I suspect this should be ``bit-field`` and should be changed to:
``A bit-field shall be a signed integer, unsigned integer, or bool``